### PR TITLE
Propagate service error when gracefully shutting down

### PR DESCRIPTION
**Pull request by FranzBusch**
# Motivation

Propagate the service error when we shutdown the group after the service threw.

### CODE REVIEW

File: Sources/ServiceLifecycle/ServiceGroup.swift [hunk 1]
```diff
@@ -338,6 +338,7 @@ public actor ServiceGroup: Sendable {
                                 group: &group,
                                 gracefulShutdownManagers: gracefulShutdownManagers
                             )
+                            return .failure(error)
```
Comment (vlm) [inline Sources/ServiceLifecycle/ServiceGroup.swift:341]: 🙇

File: Sources/ServiceLifecycle/ServiceGroup.swift [hunk 2]
```diff
@@ -338,6 +338,7 @@ public actor ServiceGroup: Sendable {
...
                         } catch {
                             return .failure(error)
```
Comment (vlm) [inline Sources/ServiceLifecycle/ServiceGroup.swift:343]: This is a different error, no?

**Review by vlm — APPROVED** [review]

File: Sources/ServiceLifecycle/ServiceGroup.swift (hunk 2 above)
Comment (FranzBusch) [reply Sources/ServiceLifecycle/ServiceGroup.swift:343]: Yes this is the error if the group fails to shutdown. What do you expect here? The error from the original service that threw or the error during shutdown?

Comment (FranzBusch) [reply Sources/ServiceLifecycle/ServiceGroup.swift]: Agreed. We are also logging that another service threw during shutdown and what error it threw.

**Pull request merged**